foundations/foundation-metrics/src/main/java/org/apache/servicecomb/foundation/metrics/publish/Metric.java

```diff
@@ -71,11 +95,28 @@ public boolean containsTag(String tagKey, String tagValue) {
   }
 
   public boolean containsTag(String... tags) {
-    for (int i = 0; i < tags.length; i += 2) {
-      if (!containsTag(tags[i], tags[i + 1])) {
-        return false;
+    if (tags.length >= 2 && tags.length % 2 == 0) {
+      for (int i = 0; i < tags.length; i += 2) {
+        if (!containsTag(tags[i], tags[i + 1])) {
+          return false;
+        }
       }
+      return true;
     }
-    return true;
+    throw new ServiceCombException("bad tags count");
+  }
+
+  private int getCharCount(String id, char c) {
+    int count = 0;
+    for (char cr : id.toCharArray()) {
+      if (cr == c) {
+        count++;
+      }
+    }
+    return count;
+  }
+
+  private boolean isCorrectId(String id) {
```

Comment:
what if the id is empty string such as "".

Reply:
Done